cmake/trilinos_ampi.patch

```diff
@@ -0,0 +1,74 @@
+diff --git a/packages/teuchos/comm/src/Teuchos_PerformanceMonitorBase.hpp b/packages/teuchos/comm/src/Teuchos_PerformanceMonitorBase.hpp
+index 6eb0b815a5..fd2bf76bf8 100644
+--- a/packages/teuchos/comm/src/Teuchos_PerformanceMonitorBase.hpp
++++ b/packages/teuchos/comm/src/Teuchos_PerformanceMonitorBase.hpp
+@@ -222,7 +222,7 @@ namespace Teuchos
+         // out of memory for storing callbacks).  We could throw an
+         // exception here in that case, but I think it's better just
+         // to let the minor memory leak happen.
+-        (void) atexit (freeTableFormat);
++        static_cast<void>(atexit(freeTableFormat));
+       }
+       TEUCHOS_TEST_FOR_EXCEPTION(
+         format_ == NULL, std::logic_error, "Teuchos::PerformanceMonitorBase::"
+@@ -305,7 +305,7 @@ namespace Teuchos
+         // out of memory for storing callbacks).  We could throw an
+         // exception here in that case, but I think it's better just
+         // to let the minor memory leak happen.
+-        (void) atexit (freeCounters);
++        static_cast<void>(atexit(freeCounters));
+       }
+       TEUCHOS_TEST_FOR_EXCEPTION(
+         counters_ == NULL, std::logic_error, "Teuchos::PerformanceMonitorBase::"
+diff --git a/packages/zoltan/src/coloring/color_test.c b/packages/zoltan/src/coloring/color_test.c
+index 465a6a97ed..47fca6fa94 100644
+--- a/packages/zoltan/src/coloring/color_test.c
++++ b/packages/zoltan/src/coloring/color_test.c
+@@ -59,7 +59,7 @@ extern "C" {
+ #include "g2l_hash.h"
+ #include "params_const.h"
+ #include "zz_util_const.h"
+-#include "graph.h"
++#include "../graph/graph.h"
```

Comment:
Should you just drop that part of the patch by setting an include path?

Reply:
It's not that simple. This collides with a Charm++ include file that is named the same way. Both include directories should be on the list of the include paths and I think the order in which they appear will decide what gets included, but the question is for what compilation unit? In this case, both are in a TPL that is out of our control. I figured, this is the safest solution, but they have not merged this PR yet: https://github.com/trilinos/Trilinos/pull/3122

Reply:
We can also rename the graph.h in Charm++. That's too generic a name for us to be putting in our include/. I opened a redmine issue for this here: https://charm.cs.illinois.edu/redmine/issues/1948

Reply:
Thanks Sam, that would be great.